cloud/pkg/edgecontroller/controller/downstream.go

```diff
@@ -453,6 +459,88 @@ func (dc *DownstreamController) syncEndpoints() {
 	}
 }
 
+func (dc *DownstreamController) syncRule() {
+	for {
+		select {
+		case <-beehiveContext.Done():
+			klog.Warning("Stop edgecontroller downstream syncRule loop")
+			return
+		case e := <-dc.rulesManager.Events():
+			klog.V(4).Infof("Get rule events: event type: %s.", e.Type)
+			rule, ok := e.Object.(*routerv1.Rule)
+			if !ok {
+				klog.Warningf("object type: %T unsupported", rule)
+				continue
+			}
+			klog.V(4).Infof("Get rule events: rule object: %+v.", rule)
+			msg := model.NewMessage("")
+			msg.SetResourceVersion(rule.ResourceVersion)
+			resource, err := messagelayer.BuildResourceForRouter(model.ResourceTypeRule, rule.Name)
+			if err != nil {
+				klog.Warningf("built message resource failed with error: %s", err)
+				continue
+			}
+			msg.Content = rule
+			switch e.Type {
+			case watch.Added:
+				msg.BuildRouter(modules.EdgeControllerModuleName, constants.GroupResource, resource, model.InsertOperation)
+			case watch.Deleted:
+				msg.BuildRouter(modules.EdgeControllerModuleName, constants.GroupResource, resource, model.DeleteOperation)
+			case watch.Modified:
+				klog.Warningf("rule event type: %s unsupported", e.Type)
+			default:
+				klog.Warningf("rule event type: %s unsupported", e.Type)
+			}
+			if err := dc.messageLayer.Send(*msg); err != nil {
+				klog.Warningf("send message failed with error: %s, operation: %s, resource: %s", err, msg.GetOperation(), msg.GetResource())
```

Comment:
better to append error to the end of a message.

Reply:
done